src/containers/Profile.jsx

```diff
@@ -0,0 +1,164 @@
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { bindActionCreators } from 'redux';
+import classnames from 'classnames';
+import {
+  selectorProfileById, selectorPasswordChange, selectorPatron, selectorInfoChange,
+} from '../selectors';
+import { profileActions } from '../actions';
+import { formatData } from '../util';
+
+class Profile extends PureComponent {
+  constructor() {
+    super();
+    this.state = {
+      password: {
```

Comment:
It is better to have plain state unless you really need to have a deep one.
Otherwise you need deep complicated stuff like this:
    this.setState({ password: { ...password, [str]: event.target.value } });